scripts/build_podspec.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2017-2019 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+set -eu
+
+function usage() {
+  echo "$0 [-u] version nio_version"
+  echo
+  echo "OPTIONS:"
+  echo "  -u: Additionally upload the podspec"
+  echo "  -f: Skip over all targets before the specified target"
+}
+
+upload=false
+skip_until=""
+while getopts ":u" opt; do
+  case $opt in
+    u)
+      upload=true
+      ;;
+    f)
+      skip_until="$OPTARG"
+      ;;
+    \?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+shift "$((OPTIND-1))"
+
+if [[ $# -eq 0 ]]; then
+  echo "Must provide target version"
+  exit 1
+fi
+
+version=$1
+
+# Current SwiftNIO Version to add as dependency in the .podspec
+nio_version=$2
+newline=$'\n'
+
+here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+tmpdir=$(mktemp -d /tmp/.build_podspecsXXXXXX)
+
+echo "Building podspec in $tmpdir"
+
+# JP - Think of a better way to handle this.. 
+#      Might be able to leverage the list_topsorted_dependencies.sh script fully
+targets=( "CNIOBoringSSL" "CNIOBoringSSLShims" "SwiftNIOSSL" )
```

Comment:
yeah, the `list_topsorted_dependencies.sh` script should do exactly this for you. Let me know if anything doesn't work

Reply:
Weird enough, running with the script actually results in more podspecs than just the libraries:
```bash
targets=( $("${here}/list_topsorted_dependencies.sh" -l -r | sed 's/^NIO/SwiftNIO/') )
```
Podspec files generated:
```
CNIOBoringSSL.podspec
CNIOBoringSSLShims.podspec
SwiftNIO.podspec
SwiftNIOConcurrencyHelpers.podspec
SwiftNIOFoundationCompat.podspec
SwiftNIOHTTP1.podspec
SwiftNIOSSL.podspec
SwiftNIOSSLPerformanceTester.podspec
SwiftNIOTLS.podspec
```

Even with the '-l' option passed 🤔

Reply:
whoops, so it also goes into targets of dependencies... That's not good :)

Reply:
Yeah, now I remember why I put that comment! 😄 I can take on that as a follow-up item? The idea behind the `list_topsorted_dependencies.sh` script is pretty clever, and I'd like to leverage that here for sure.

Reply:
added https://github.com/apple/swift-nio/pull/1420 to fix this

Reply:
Updated here as well 👍